src/Glpi/Asset/CustomFieldType/AbstractType.php

```diff
@@ -70,16 +71,13 @@ public function getOptions(): array
     {
         return [
             new BooleanOption($this->custom_field, 'full_width', __('Full width'), false),
-            new BooleanOption($this->custom_field, 'readonly', __('Readonly'), false),
             new BooleanOption($this->custom_field, 'required', __('Mandatory'), false),
-            new BooleanOption($this->custom_field, 'disabled', __('Disabled'), false), // Not exposed in the UI. Only used in field order preview
+            new ProfileRestrictOption($this->custom_field, 'readonly', __('Readonly'), false),
+            new ProfileRestrictOption($this->custom_field, 'hidden', __('Hidden'), false),
```

Comment:
```suggestion
            new ProfileRestrictOption($this->custom_field, 'readonly', __('Readonly for these profiles'), false),
            new ProfileRestrictOption($this->custom_field, 'hidden', __('Hidden for these profiles'), false),
```

Additionally, I suggest to enlarge the modal for native fields (same as the one for custom fields)